File: src/main/java/com/robohorse/robopojogenerator/generator/common/JsonItem.java
package com.robohorse.robopojogenerator.generator.common;

import org.json.JSONObject;

/**
 * Created by vadim on 22.01.17.
 */
public class JsonItem {
    private String key;
    private JSONObject jsonObject;

    public JsonItem(JSONObject jsonObject, String key) {
        this.key = key;
        this.jsonObject = jsonObject;
    }

    public String getKey() {
        return key;
    }

    public JSONObject getJsonObject() {
        return jsonObject;
    }
}

File: src/main/kotlin/com/robohorse/robopojogenerator/generator/consts/common/JsonItem.kt
package com.robohorse.robopojogenerator.generator.consts.common

import org.json.JSONArray
import org.json.JSONObject

sealed class JsonModel(
    open val key: String,
    val keySet:Set<String>
) {
    data class JsonItem(
        override val key: String,
        val jsonObject: JSONObject
    ) : JsonModel(key, jsonObject.keySet())

    data class JsonItemArray(
        override val key: String,
        val jsonObject: JSONArray
    ) : JsonModel(key, emptySet())
}
